frontend/test/metabase/scenarios/embedding/embedding-full-app.cy.spec.js

```diff
@@ -97,6 +114,181 @@ describe("scenarios > embedding > full app", () => {
+
+        cy.findByRole("banner").within(() => {
+          cy.findByText(/Our analytics/i).should("not.be.visible");
+          cy.findByTestId("main-logo").should("be.visible");
+        });
+      });
+
+      it("should hide side nav toggle icon on hover", () => {
+        // default logo: true
+        visitQuestionUrl({ url: "/question/1", qs: { side_nav: false } });
+        cy.findByRole("banner").within(() => {
+          cy.findByText(/Our analytics/i).should("be.visible");
+          cy.findByTestId("main-logo").should("be.visible");
+        });
+
+        cy.findByTestId("sidebar-toggle").should("not.exist");
+      });
+
+      it("should always show side nav toggle icon when logo is hidden", () => {
+        visitQuestionUrl({
+          url: "/question/1",
+          qs: { side_nav: true, logo: false },
+        });
+        cy.findByRole("banner").within(() => {
+          cy.findByText(/Our analytics/i).should("be.visible");
+          cy.findByTestId("main-logo").should("not.be.visible");
+          cy.icon("sidebar_closed").should("be.visible");
+        });
+        // Non-hovered color
+        cy.findByTestId("sidebar-toggle")
+          .children()
+          .should("have.css", "color", toRgbString(color("text-medium")));
+        // hovered color
+        cy.findByTestId("sidebar-toggle")
+          .realHover()
+          .children()
+          .should("have.css", "color", toRgbString(color("brand")));
+
+        cy.findByTestId("sidebar-toggle").click();
+        cy.findByText(/Add your own data/i).should("be.visible");
+
+        cy.findByRole("banner").within(() => {
+          cy.findByText(/Our analytics/i).should("not.be.visible");
+          cy.findByTestId("main-logo").should("not.be.visible");
+          cy.icon("sidebar_open").should("be.visible");
+        });
+      });
+
+      it("should not show either logo or side nav toggle button at all", () => {
+        visitQuestionUrl({
+          url: "/question/1",
+          qs: { side_nav: false, logo: false },
+        });
+        cy.findByRole("banner").within(() => {
+          cy.findByText(/Our analytics/i).should("be.visible");
+          cy.findByTestId("main-logo").should("not.be.visible");
+          cy.icon("sidebar_closed").should("not.exist");
+        });
+
+        cy.findByTestId("sidebar-toggle").should("not.exist");
+      });
+
+      it("should hide main header when there's nothing to display there", () => {
+        visitQuestionUrl({
+          url: "/question/1",
+          qs: { side_nav: false, logo: false, breadcrumbs: false },
+        });
+        cy.findByRole("banner").should("not.exist");
+        cy.findByTestId("main-logo").should("not.exist");
+        cy.icon("sidebar_closed").should("not.exist");
+        cy.findByTestId("sidebar-toggle").should("not.exist");
+      });
+    });
+
+    describe("mobile logo", () => {
+      beforeEach(() => {
+        cy.viewport("iphone-x");
```

Comment:
Test mobile view also, since we basically render a different component entirely.

on desktop: https://github.com/metabase/metabase/blob/master/frontend/src/metabase/nav/components/AppBar/AppBarLarge.tsx
on mobile: https://github.com/metabase/metabase/blob/master/frontend/src/metabase/nav/components/AppBar/AppBarSmall.tsx